sycl/include/sycl/detail/kernel_name_str_t.hpp

```diff
@@ -17,10 +18,19 @@ namespace detail {
 using KernelNameStrT = std::string_view;
 using KernelNameStrRefT = std::string_view;
 using ABINeutralKernelNameStrT = detail::string_view;
+
+inline KernelNameStrT toKernelNameStrT(ABINeutralKernelNameStrT str) {
```

Comment:
Do the `str` argument constness need to differ between the definitions? I would prefer we keep them the same, even if it means removing `const` in the variant below. If it gets removed with breaking changes, I would argue it's better to have it the same and avoid the potential hidden faulty const use-cases we could have.

Reply:
Sorry, probably I do not understand. Are you for passing `detail::string_view` by const reference, like this?
```
inline KernelNameStrT toKernelNameStrT(const ABINeutralKernelNameStrT &str) {
#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
  return std::string_view(str);
#else
  return str.data();
#endif
}
```